### ISSUE TITLE
Diagram Factory / Generator for easy adding new diagrams

### ISSUE THREAD

**Issue by nirname** [mermaid-js/mermaid#4606]
To simplify new diagrams creation process and force developers to create new diagrams in a standard way we need a template that would allow us to generate all required files for the diagram

The structure of generated files should match those in discussion #4499

**Comment by Yokozuna59** [comment]
I almost have the list of the files need to be created/modified when creating new diagram, but it's need organizing and more details for what to be added/modified.

Originally, I was planning to create it when finalizing the stuff discussed in #4499.

**Comment by sidharthv96** [comment]
`mermaid-diagram-example` might actually be a bad example. We should properly type and refactor it first before using it.

**Comment by Yokozuna59** [comment]
> `mermaid-diagram-example` might actually be a bad example. We should properly type and refactor it first before using it.

We should also export some types from `packages/mermaid` to be able to create other diagrams similar to `mermaied-zenuml`, currently it's not fully typed.

**Comment by Yokozuna59** [comment]
That's also my intention, but I was referring to where we should store those files.
Of course, it doesn't make sense to create file in a string variable, we need to store those files some where that the TS language server and prettier could run and throw errors. That's why I suggested using the [`packages/mermaid-example-diagram`](https://github.com/mermaid-js/mermaid/tree/develop/packages/mermaid-example-diagram) as a template.

And when running the command it would add it to [`packages/mermaid/src/diagrams`](https://github.com/mermaid-js/mermaid/tree/develop/packages/mermaid/src/diagrams).

**Comment by nirname** [comment]
Yep, exactly. It doesn't have to be `mermaid-example-diagram` either, it can be another package like `packages/diagram-template`. We could rework `mermaid-example-diagram` but its name does not reflect its potential purpose, so we probably could rename it or create another package for this task.

**Comment by Yokozuna59** [comment]
Maybe renaming could be better, it doesn't has any other use right now.

**Comment by Yokozuna59** [comment]
The script should handle two type of diagram templates, external and internal.

- External:

  Similar to [`mermaid-zenuml`](https://github.com/mermaid-js/mermaid/tree/07f460a1846d22e54016c25862b9eb42642583ef/packages/mermaid-zenuml) and [`mermaid-example-diagram`](https://github.com/mermaid-js/mermaid/tree/07f460a1846d22e54016c25862b9eb42642583ef/packages/mermaid-example-diagram).

- Internal:

  Should be inside [`packages/mermaid/src/diagrams`](https://github.com/mermaid-js/mermaid/tree/07f460a1846d22e54016c25862b9eb42642583ef/packages/mermaid/src/diagrams), so similar to most existing diagrams.
